index.js

```diff
@@ -12,7 +12,58 @@ function setMultipart (req, done) {
   done()
 }
 
+function attachToBody (options, req, reply, next) {
+  req[kMultipart] = true
+
+  const defaultConsumer = (field, file, filename, encoding, mimetype) => {
```

Comment:
Can you make this a top level function? I think we should also pass in the body object.

Reply:
I leave the body out from the parameter (and used the clojure instead) because I was thinking that could be dangerous pass also the body object because the user could manipulate it in a wrong pattern (without using hooks for example).

Am I too careful?

Reply:
The problem with that approach is that a user cannot implement this function with the public API. They can only give in a top-level function, while you are creating a closure here. The functionality should be identical.

A simplest thing would be to expose the body.